jaxopt/_src/levenberg_marquardt.py

```diff
@@ -326,6 +329,9 @@ def update(self, params, state: NamedTuple, *args, **kwargs) -> base.OptStep:
     # Current value of the loss function F=0.5*||f||^2.
     loss_curr = state.loss
 
+    # TODO: materialize_jac currently silently use LU factorization. Needs to be 
+    # updated to use direct linear solve with passing the matrix.
+
     # For geodesic acceleration, we calculate  jtrpp=JT * r",
     # where J is jacobian and r" is second order directional derivative.
     if self.materialize_jac:
```

Comment:
there is one important difference, for the case of `materialize_jac=True`, we are materializing Jacobian, IOW, calculating the gradients using `jacfwd`. If this option is false, all variables that depend on the Jacobian like `J^T.J`, are obtained with `jvp` and `vjp` tricks, that's why you see the operators like `_jt_op`, `_jtj_op` etc. I will update the documentation so it becomes more clear.

Reply:
Ok, I see. The wording ``materialize_jac=True'' is confusing since even if it is false, as long as the solver is not a matrix free solver like cg or gmres (or any solver in linear_solver that does not use '_materialize_array'), the jtj will be materialized and there could be a substantial overhead (materializing the matrix can be as costly as inverting it a priori, especially for relatively low precision). I would add in the postprocess a warning if materialize_jax is false and the solver is not a matrix-free solver.
By the way the vjp is computed at several places (in _jt_op and _jtj_op). It could be shared to save some memory.

Reply:
We may rethink the use of vjp for another CL I think. The warning is great.